src/databricks/labs/lakebridge/contexts/application.py

```diff
@@ -138,3 +138,11 @@ def upgrades(self):
     def analyzer(self):
         is_debug = logger.getEffectiveLevel() == logging.DEBUG
         return LakebridgeAnalyzer(self.current_user, self.prompts, is_debug)
+
+    def add_user_agent_extra(self, key: str, value: str) -> None:
+        with_user_agent_extra(key, value)
+        new_config = self._ws.config.with_user_agent_extra(key, value)
```

Comment:
One thing I just noticed is that `with_user_agent_extra()` modifies the configuration rather than returning a new instance. As such I think I'd prefer to see this as:
```suggestion
        new_config = self._ws.config.copy().with_user_agent_extra(key, value)
```